sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -129,7 +229,15 @@ export namespace ConnectionContext {
       );
     };
 
-    const disconnected: OnAmqpEvent = async (context: EventContext) => {
+    const onDisconnected: OnAmqpEvent = async (context: EventContext) => {
+      if (isDisconnecting) {
+        return;
+      }
+      isDisconnecting = true;
+      waitForDisconnectPromise = new Promise((resolve) => {
```

Comment:
It looks like the `isDisconnecting` variable is basically the same as `waitForDisconnectPromise != null`. Worth it to remove and just have the one variable to track if we're disconnecting?

Reply:
Yeah...I have a check in readyToOpenLink() that checks `isDisconnecting`, which just felt easier to understand than `if (waitForDisconnectPromise)` or `if (Boolean(waitForDisconnectPromise))`, but  it might be better to have one source of truth.

Reply:
Updated!